src/components/WidthRestrictor/BlockWidthRestrictor/__tests__/BlockWidthRestrictor.spec.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, RenderResult } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { BlockWidthRestrictor } from '../BlockWidthRestrictor';
+
+describe('BlockWidthRestrictor', () => {
+    let view: RenderResult;
+
+    it('should render nothing if no children are provided', () => {
+        view = render(<BlockWidthRestrictor />);
+
+        expect(view.container).toBeEmptyDOMElement();
+    });
+
+    it('should render correctly', () => {
+        view = render(<BlockWidthRestrictor>Some children</BlockWidthRestrictor>);
+
+        expect(view.container).toMatchSnapshot();
+        expect(view.container).toHaveTextContent('Some children');
+        expect(view.container.children[0].tagName).toBe('DIV');
+    });
+
+    it('should render correctly when element type is defined', () => {
+        view = render(<BlockWidthRestrictor As="nav">Some children</BlockWidthRestrictor>);
+
+        expect(view.container.children).toHaveLength(1);
+        expect(view.container.children[0].tagName).toBe('NAV');
```

Comment:
s/a

Reply:
Since the intent here is to verify that the `As` prop indeed effects the HTML tag that is rendered, it is a good test I think. If you can look up what the implicit role for `nav` is and use `getByRole` instead, that would be better.

I'm not sure of the usefulness of the previous line though...

Reply:
:+1: